examples/rbd/template-csi-rbdplugin-patch.yaml

```diff
@@ -0,0 +1,30 @@
+---
+# This is a patch to the existing daemonset deployment of CSI rbdplugin.
+#
+# This is to be used when using `clusterID` instead of monitors or
+# monValueFromSecret in the StorageClass to specify the Ceph cluster to
+# provision storage from, AND when the value of `--configroot` option to the
+# CSI pods is NOT "k8s_objects".
+#
+# This patch file, patches in the specified secret for the 'clusterID' as a
+# volume, instead of the Ceph CSI plugin actively fetching and using kubernetes
+# secrets.
+#
+# NOTE: Update csi-rbdplugin-provisioner StatefulSet as well with similar patch
+# Post substituting the <cluster-id> in all places execute,
+#   `kubectl patch daemonset csi-rbdplugin --patch\
+#         "$(cat template-csi-rbdplugin-patch.yaml)"`
+#   to patch the statefulset deployment.
+spec:
```

Comment:
as the codes changes work with the current templates.  
can we add this to the existing yaml files instead of asking users to patch?

IMO in fresh deployment, if we do a patch people we normally think we are doing some kind of hack.

Reply:
The comment above reads:

This is to be used when using `clusterID` instead of monitors or
monValueFromSecret in the StorageClass to specify the Ceph cluster to
provision storage from, **AND** when the value of `--configroot` option to the
CSI pods is **NOT "k8s_objects"."**

So in a fresh installation when using the provided pod yaml (which default configroot to "k8s_object") these templates need not be used at all.

As a result these templates are also not discussed in other documentation. These are here for reference in case someone does not want to use the configroot as "k8s_object".

Is there a clearer way you would like this documented?